### PR TITLE
fix(autoscaling): Allow adding AutoScalingGroup to multiple target groups

### LINKED ISSUE #5667: Cannot add AutoScalingGroup to more than one Target Group
When attempting to assign an ASG to multiple target groups, CDK errors out with `Cannot add AutoScalingGroup to 2nd Target Group`

CFN supports passing in a list of Target Group ARNs to an autoscaling group - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html#cfn-as-group-targetgrouparns

### Use Case

This feature is necessary if anybody would like to assign multiple load balancers to an ASG

### Proposed Solution

Remove the check here - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts#L559

Figure out any other functions that are effected and come up with workarounds.  Rico thinks there could be issues with scaleOnRequestCount()

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by azatoth**
While this doesn't solve the underlying problem, which will require an API breakage in `scaleOnRequestCount()` (see
https://github.com/aws/aws-cdk/issues/5667#issuecomment-636657482).

As `scaleOnRequestCount()` seems to be the only part that is affected by this, we'll only issue a validation error if we have multiple `targetGroupArns` and `scaleOnRequestCount()` has been called.

closes #5667

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by azatoth** [comment]
> The pull request linter fails with the following errors:
> 
> ```
> ❌ Fixes must contain a change to an integration test file and the resulting snapshot.
> ```
> 
> PRs must pass status checks before we can provide a meaningful review.

I don't think an update to integration is needed as we are not changing anything in what CFN is created.

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts [hunk 1]
```diff
@@ -1765,6 +1765,16 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
       };
     }
   }
+
+
+  private validateTargetGroup(): string[] {
+    const errors = new Array<string>();
+    if (this.hasSetScaleOnRequest && this.targetGroupArns.length > 1) {
+      errors.push('Cannon use multiple target groups if `setScaleOnRequest()` is being used.');
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:1773]: `setScaleOnRequest()` should be `scaleOnRequestCount()`, no? `scaleOnRequestCount()` enables `hasSetScaleOnRequest()`, I don't see a `setScaleOnRequest()` method in the `auto-scaling-group.ts`

File: packages/@aws-cdk/aws-autoscaling/test/auto-scaling-group.test.ts [hunk 2]
```diff
@@ -1864,6 +1865,43 @@ describe('auto scaling group', () => {
...
 
   });
+
+  test('Should validate multiple target groups i conjunction with `setScaleOnRequest()`', () => {
+    const stack = new cdk.Stack(undefined, 'MyStack', { env: { region: 'us-east-1', account: '1234' } });
+
+
+    const vpc = mockVpc(stack);
+
+    const alb = new ApplicationLoadBalancer(stack, 'alb', {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-autoscaling/test/auto-scaling-group.test.ts:1875]:
```suggestion
    const stack = new cdk.Stack(undefined, 'MyStack', { env: { region: 'us-east-1', account: '1234' } });
    const vpc = mockVpc(stack);
    const alb = new ApplicationLoadBalancer(stack, 'alb', {
```

File: packages/@aws-cdk/aws-autoscaling/test/auto-scaling-group.test.ts [hunk 3]
```diff
@@ -1864,6 +1865,43 @@ describe('auto scaling group', () => {
...
+    asg.attachToApplicationTargetGroup(atg1);
+    asg.attachToApplicationTargetGroup(atg2);
+
+    expect(asg.node.validate()).toEqual([]);
+
+    asg.scaleOnRequestCount('requests-per-minute', { targetRequestsPerMinute: 60 });
+
+    expect(asg.node.validate()).toContainEqual('Cannon use multiple target groups if `setScaleOnRequest()` is being used.');
+
+  });
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-autoscaling/test/auto-scaling-group.test.ts:1904]:
```suggestion
    expect(asg.node.validate()).toContainEqual('Cannon use multiple target groups if `setScaleOnRequest()` is being used.');
  });
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
This looks really good! Could we add a unit test for the positive case too?

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts (hunk 1 above)
Comment (azatoth) [reply packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts]: Oops, the "set" part in the variable meant to reflect that the method had been called; Seems I transposed that to method name when writing the error message. Changed in 1f3f3f00145308a2e81106297d2277da4b539241

**Comment by azatoth** [comment]
> This looks really good! Could we add a unit test for the positive case too?

The test for the positive case was kinda in the test already; In any case, I did split the test into two separate tests in db1f0dee69e9587a147abf77154f9d50ba29dc3a to be more explicit.

**Review by comcalvi — APPROVED** [review]

**Pull request merged**